test/tcpBackoff.js

```diff
@@ -0,0 +1,28 @@
+'use strict'
+/* eslint-env node, mocha */
+
+const TcpConnection = require('../lib/TcpConnection')
+const { ExponentialStrategy } = require('backoff')
+const { expect } = require('chai')
+
+test('tcp backoff', function testTcpBackoff (done) {
+  const exponentialStrategy = new ExponentialStrategy({
+    initialDelay: 10,
+    factor: 10 // 10, 100, 1000, 2000...
+  })
+  const tcpConnection = TcpConnection({
+    address: '127.0.0.1',
+    port: 0,
+    reconnect: true,
+    backoffStrategy: exponentialStrategy,
+    onSocketClose: () => {
+      // noop
+    }
+  })
+  setTimeout(() => {
+    tcpConnection.end('', 'utf8', () => done())
+    // initial, 10, 100
+    const nextBackoffDelay = exponentialStrategy.next()
+    expect(nextBackoffDelay).to.be.gt(100)
+  }, 500)
```

Comment:
please use fake timers to avoid test flakyness.

Reply:
We need to wait because the TCP connection will try to connect, will fail to connect and then will close. We will receive a close event and call `reconnect`. The best I can do is count the number of times `onSocketClose` was called and check if the backoff delay has increased?
I'm not sure how fake timers would help here? Am I missing something?

Reply:
I did just that, let me know if it's OK